ethereum/core/src/test/java/tech/pegasys/pantheon/ethereum/privacy/EnclaveHandlerTest.java

```diff
@@ -0,0 +1,224 @@
+import tech.pegasys.pantheon.enclave.types.DeletePrivacyGroupRequest;
+import tech.pegasys.pantheon.enclave.types.FindPrivacyGroupRequest;
+import tech.pegasys.pantheon.enclave.types.PrivacyGroup;
+import tech.pegasys.pantheon.enclave.types.ReceiveRequest;
+import tech.pegasys.pantheon.enclave.types.ReceiveResponse;
+import tech.pegasys.pantheon.enclave.types.SendRequest;
+import tech.pegasys.pantheon.enclave.types.SendRequestPantheon;
+import tech.pegasys.pantheon.enclave.types.SendResponse;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletionException;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnclaveHandlerTest {
+  @Mock private EnclaveUser enclaveUser1;
+  @Mock private EnclaveUser enclaveUser2;
+  @Mock private EnclaveUser enclaveUser3;
+  @Mock private EnclaveUser unknownEnclaveUser;
+  @Mock private Enclave enclave1;
+  @Mock private Enclave enclave2;
+  @Mock private Enclave enclave3;
+  private EnclaveHandler enclaveHandler;
+
+  @Before
+  public void setup() {
+    final Map<EnclaveUser, Enclave> enclaves =
+        ImmutableMap.of(enclaveUser1, enclave1, enclaveUser2, enclave2, enclaveUser3, enclave3);
+    enclaveHandler = new EnclaveHandler(enclaves);
+  }
+
+  @Test
+  public void sendsDataToUsersOrion() throws Exception {
+    final SendRequest sendRequest = new SendRequestPantheon("payload", "from", "privacyGroup");
+    when(enclave1.send(sendRequest)).thenReturn(new SendResponse("abc123"));
+
+    assertThat(enclaveHandler.sendData(enclaveUser1, sendRequest))
+        .isEqualToComparingFieldByField(new SendResponse("abc123"));
+    verify(enclave1).send(sendRequest);
+    verifyZeroInteractions(enclave2);
+  }
+
+  @Test
+  public void sendDataShouldFailIfNoEnclaveUserConfiguredForAnEnclave() {
+    final SendRequest sendRequest = new SendRequestPantheon("payload", "from", "privacyGroup");
+    assertThatThrownBy(() -> enclaveHandler.sendData(unknownEnclaveUser, sendRequest))
+        .isInstanceOf(EnclaveException.class);
+    verifyZeroInteractions(enclave1, enclave2, enclave3);
+  }
+
+  @Test
+  public void retrievesDataForEnclaveUser() throws Exception {
+    final ReceiveRequest receiveRequest = new ReceiveRequest("abc123", "user2PublicKey");
+
+    when(enclave1.receive(any(ReceiveRequest.class))).thenThrow(EnclaveException.class);
+    when(enclave2.receive(any(ReceiveRequest.class))).thenThrow(EnclaveException.class);
+    when(enclave3.receive(receiveRequest)).thenReturn(new ReceiveResponse(null, "privacyGroup"));
+
+    assertThat(enclaveHandler.retrieveData(receiveRequest).getPrivacyGroupId())
+        .isEqualTo("privacyGroup");
+    verify(enclave1).receive(any());
```

Comment:
should it be receive(any()) or should it be receive(receiveRequest)?
If so - this applies to other "verify" checks as well.

Reply:
should be receive(receiveRequest) for the verify. fixed everywhere hopefully.